### PR TITLE
Fix retries

### PR DESCRIPTION
**Pull request by trask**
I'm not seeing any retries.

I'm very new to rust, so probably missing how it's supposed to work, but this change made retries work for me.

### CODE REVIEW

File: lychee-lib/src/retry.rs [hunk 1]
```diff
@@ -71,10 +71,9 @@ impl RetryExt for reqwest::Error {
             } else {
                 false
             }
+        } else if let Some(status) = self.status() {
+            status.should_retry()
         } else {
-            // We omit checking if error.is_status() since we check that already.
```
Comment (mre) [inline lychee-lib/src/retry.rs:75]: Can we keep this comment?

**Comment by mre** [comment]
Thanks for the PR! Makes sense.
Can you add a test to check that it works?
Ideally, a unit test in the same file and an integration test in https://github.com/lycheeverse/lychee/blob/master/lychee-bin/tests/cli.rs. The integration test can be modeled after the existing tests. For example, you can use a mock http server as in this test: https://github.com/lycheeverse/lychee/blob/10056dc5e939be40a697bf30fecb9a9fdfec2d95/lychee-bin/tests/cli.rs#L547-L557

The unit test would be about creating a new status object and checking that it returns the correct result when calling `should_retry()` on it. For some reason, we don't have unit tests for that yet (hence why it's probably broken), but it's a great opportunity to change that. 😉

**Comment by trask** [comment]
unit test and integration test added!

**Comment by mre** [comment]
Brilliant! Thanks for the quick turnaround and for the clean implementation.

**Pull request merged**